tasks/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml

```diff
@@ -0,0 +1,261 @@
+        
+            sbom_version="$(jq -r ".specVersion" "$sbom_path")"
+            supportedCycloneDxVersion=$(params.supportedCycloneDxVersion)
+            version_lesser_equal "$sbom_version" "$supportedCycloneDxVersion"
+        
+            printf "%s" "cyclonedx-json" > "${sbom_path}.sbom_format"
+        
+          elif [ "$sbom_format" == "SPDX" ]; then
+            echo "Found SPDX SBOM: $file_relpath"
+        
+            sbom_version="$(jq -r '.spdxVersion | sub("SPDX-"; "")' "$sbom_path")"
+            supportedSpdxVersion=$(params.supportedSpdxVersion)
+            version_lesser_equal "$sbom_version" "$supportedSpdxVersion"
+        
+            printf "%s" "spdx-json" > "${sbom_path}.sbom_format"
+        
+          else
+            echo "$file_relpath: not a valid SBOM (CycloneDX or SPDX)"
+            continue
+          fi
+        done
+        echo "Found $(find "$workdir" -name "*.json" | wc -l) SBOMs"
+
+    # Needs syft, which is in a different image => has to be a separate step
+    - name: convert-sboms-if-needed
+      image: registry.redhat.io/rh-syft-tech-preview/syft-rhel9:1.0.1@sha256:27c268d678103a27b6964c2cd5169040941b7304d0078f9727789ffb8ffba370
+      script: |
+        #!/usr/bin/env bash    
+        set -o errexit -o pipefail -o nounset
+    
+        skipUploadingSbom=$(params.skipUploadingSbom)
+        if [[ "$skipUploadingSbom" == "true" ]]; then
+          exit 0
+        fi
+  
+        # Return zero matches when a glob doesn't match rather than returning the glob itself
+        shopt -s nullglob
+  
+        for sbom_path in "$workdir"/*.json; do
+          conversion_attr="${sbom_path}.convert_to_version"
+          sbom_format="$(cat "${sbom_path}.sbom_format")"
+  
+          if [[ -f "$conversion_attr" ]]; then
+            sbom_version="$(cat "$conversion_attr")"
+            original_sbom_path="$(realpath "$sbom_path")"
+            original_sbom_relpath="$(realpath "$sbom_path" --relative-base="$sbomsDir")"
+  
+            echo "Converting $original_sbom_relpath to $sbom_format $sbom_version"
+            syft convert "$original_sbom_path" -o "${sbom_format}@${sbom_version}=${sbom_path}.supported_version"
+          else
+            # Just duplicate the symlink, the original SBOM already has a supported CDX version
+            cp --no-dereference "$sbom_path" "${sbom_path}.supported_version"
+          fi
+        done
+
+    - name: upload-sboms
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      volumeMounts:
+        - name: atlas-secret
+          mountPath: /secrets/
+      script: |
+        #!/usr/bin/env bash    
+        set -o errexit -o pipefail -o nounset
+
+        skipUploadingSbom=$(params.skipUploadingSbom)
+        if [[ "$skipUploadingSbom" == "true" ]]; then
+          exit 0
+        fi
+  
+        shopt -s nullglob
+        sboms_to_upload=("$workdir"/*.json)
```

Comment:
I don't know how this is supposed to work - how `/tekton/home` is preserved among the steps or if you even have access to write there. It might. But I would recommend using an `emptyDir` for sharing data among steps.

See e.g. here: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-rpm-data-to-pyxis/push-rpm-data-to-pyxis.yaml#L37